tools/integration-test/src/tests/query_packet.rs

```diff
@@ -0,0 +1,149 @@
+use ibc_relayer::chain::counterparty::{channel_on_destination, pending_packet_summary};
+use ibc_relayer::link::{Link, LinkParameters};
+
+use ibc_test_framework::prelude::*;
+use ibc_test_framework::relayer::channel::query_identified_channel_end;
+use ibc_test_framework::relayer::connection::query_identified_connection_end;
+use ibc_test_framework::util::random::random_u64_range;
+
+#[test]
+fn test_query_packet_pending() -> Result<(), Error> {
+    run_binary_channel_test(&QueryPacketPendingTest)
+}
+
+pub struct QueryPacketPendingTest;
+
+impl TestOverrides for QueryPacketPendingTest {
+    fn modify_relayer_config(&self, config: &mut Config) {
+        // Disabling clear_on_start should make the relayer not
+        // relay any packet it missed before starting.
+        config.mode.packets.clear_on_start = false;
+        config.mode.packets.clear_interval = 0;
+    }
+
+    fn should_spawn_supervisor(&self) -> bool {
+        false
+    }
+
+    // Unordered channel: will permit gaps in the sequence of relayed packets
+    fn channel_order(&self) -> Order {
```

Comment:
We can remove this method since the trait implementation defaults to `Order::Unordered`.